### PR TITLE
Implement Cypress Functional Testing

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR introduces Cypress testing for the Kitchen Sink form, containing a sample suite of our simple components.
- Restructure existing external tests to new folder hierarchy
- Implement insulated kitchen sink functional test in Cypress
- Add documentation and hooks for future pipeline/CI integration

[SHOWCASE-216](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-216)

## Types of changes

New feature (non-breaking change which adds functionality)
Documentation (non-breaking change with enhancements to documentation)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-206

File: README.md [hunk 1]
```diff
@@ -22,6 +22,11 @@ Create, edit and publish forms.
     components/                - Form.io Custom Components Library
     load-test/                 - Standalone CHEFS load testing utility
```
Comment (loneil) [inline README.md:23]: Can remove this line from the readme?

Comment (jujaga) [reply README.md:23]: Oh whoops nice catch! Fixed.

**Review by MichaelChung123 — APPROVED** [review]
Looks good. Tested the Cypress functionality on my local, approving.

**Review by TimCsaky — APPROVED** [review]

**Pull request merged**